s3prl/downstream/specaug.py

```diff
@@ -208,10 +227,23 @@ def mask_along_axis(self, spec, spec_lengths):
         B = spec.shape[0]
         # D = Length or Freq
         D = spec.shape[self.dim]
+        T = self.mask_width_range[1]
+        # Adaptive_SpecAugment
+        if self.dim == 1 & self.adaptive :
+          if self.adaptive_number_ratio > 0:
+            self.num_mask = min(int(self.adaptive_number_ratio * D), self.max_n_time_masks)
```

Comment:
Yes, I think we should change the code as your suggestion, since I may misconstrue the difference between the `self.num_mask` and `num_mask`. Thanks for your correction!

Reply:
> It is a little bit weird to me that an attribute (`self.num_mask`) is dynamically changed along with the most recent input. Could we change to `num_mask = min(int(self.adaptive_number_ratio * D), self.max_n_time_masks)` and use `num_mask` (instead of `self.num_mask`) in the code below?

I have followed you recommendation, using variables instead of directly changing `self.num_mask`.